knowrob_srdl/prolog/knowrob/urdf_parser.plt

```diff
@@ -0,0 +1,400 @@
+/*
+  Copyright (C) 2018 Georg Bartels
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are met:
+      * Redistributions of source code must retain the above copyright
+        notice, this list of conditions and the following disclaimer.
+      * Redistributions in binary form must reproduce the above copyright
+        notice, this list of conditions and the following disclaimer in the
+        documentation and/or other materials provided with the distribution.
+      * Neither the name of the <organization> nor the
+        names of its contributors may be used to endorse or promote products
+        derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+:- begin_tests(urdf_parser).
+
+:- use_module('urdf_parser').
+:- use_module(library('roscpp')).
+
+test(load_urdf_file_pr2) :-
+  ros_package_path('knowrob_srdl', X),
+  atom_concat(X, '/urdf/pr2_for_unit_tests.urdf', Filename),
+  load_urdf_file(Filename).
+
+test(load_non_existent_urdf, fail) :-
+  load_urdf_file('foo.urdf').
+
+test(robot_name_pr2) :-
+  robot_name(pr2).
+
+test(root_link_name_pr2) :-
+  root_link_name(base_footprint).
+
+test(joint_names_pr2) :-
+  joint_names(Names),
+  Names == [base_bellow_joint, base_footprint_joint, base_laser_joint,
```

Comment:
what the heck is github doing here? can't really review this file through github